app/controllers/works_controller.rb

```diff
@@ -11,54 +11,95 @@ def root
   end
 
   def index
-    @works_by_category = Work.to_category_hash
+    if @login_user
+      @works_by_category = Work.to_category_hash
+    else
+      flash[:status] = :failure
+      flash[:result_text] = "Must be logged in to view page."
+      redirect_to root_path
+    end
   end
 
   def new
-    @work = Work.new
+    if @login_user
+      @work = Work.new
+    else
+      flash[:status] = :failure
```

Comment:
You have this check for whether the user is logged in repeated many times. Could you DRY this up with a controller filter?